### PR TITLE
Fix the autocomplete

### PR DESCRIPTION
**Pull request by TsvetanMilanov**
Fix http://teampulse.telerik.com/view#item/292768

### CODE REVIEW

**Comment by justcodebuilduser** [comment]
:heart:

File: services/commands-service.ts [hunk 1]
```diff
@@ -248,54 +248,73 @@ export class CommandsService implements ICommandsService {
...
+				if (command) {
 					let completionData = command.completionData;
-					if(completionData) {
+					if (completionData) {
 						return tabtab.log(completionData, data);
+					} else {
+						return this.logChildrenCommandsNames(commandName, commands, tabtab, data);
+					}
+				} else if (childrenCommands) {
+					let commandsToMap = _.reject(childrenCommands, (children: string) => children[0] === '*');
```
Comment (rosen-vladimirov) [inline services/commands-service.ts]: this will reject the default commands, is this expected

Comment (rosen-vladimirov) [inline services/commands-service.ts]: `commandsToMap` -> `nonDefaultSubCommands`

File: services/commands-service.ts [hunk 2]
```diff
@@ -248,54 +248,73 @@ export class CommandsService implements ICommandsService {
...
+					let mappedChildrenCommands: string[] = [];
+
+					if (data.words !== line.length) {
+						mappedChildrenCommands = commandsToMap.map((commandToMap: string) => {
+							return commandToMap.substring(0, commandToMap.indexOf("|") !== -1 ? commandToMap.indexOf("|") : commandToMap.length);
```
Comment (rosen-vladimirov) [inline services/commands-service.ts]:
```TypeScript
mappedChildrenCommands = commandsToMap.map((commandToMap: string) => {
 		let pipePositon = commandToMap.indexOf("|");
 		return commandToMap.substring(0, pipePositon  !== -1 ? pipePositon  : commandToMap.length);
});

File: services/commands-service.ts [hunk 3]
```diff
@@ -248,54 +248,73 @@ export class CommandsService implements ICommandsService {
...
 					let completionData = command.completionData;
-					if(completionData) {
+					if (completionData) {
 						return tabtab.log(completionData, data);
+					} else {
+						return this.logChildrenCommandsNames(commandName, commands, tabtab, data);
+					}
+				} else if (childrenCommands) {
+					let commandsToMap = _.reject(childrenCommands, (children: string) => children[0] === '*');
+					let mappedChildrenCommands: string[] = [];
```
Comment (rosen-vladimirov) [inline services/commands-service.ts]: `sanitizedChildrenCommands`

**Comment by justcodebuilduser** [comment]
:heart:

**Pull request merged**